lib/types.nix

```diff
@@ -453,6 +453,16 @@ rec {
       functor = (defaultFunctor name) // { wrapped = elemType; };
     };
 
+    functionTo = elemType: mkOptionType {
```

Comment:
:+1: good name, leaves room for `function` with a typed input, in case that becomes relevant someday.